base/linalg/triangular.jl

```diff
@@ -1867,48 +1867,53 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
 end
 logm(A::LowerTriangular) = logm(A.').'
 
-function sqrtm{T}(A::UpperTriangular{T})
-    n = checksquare(A)
+function sqrtm(A::UpperTriangular)
     realmatrix = false
     if isreal(A)
         realmatrix = true
-        for i = 1:n
+        for i = 1:checksquare(A)
             if real(A[i,i]) < 0
                 realmatrix = false
                 break
             end
         end
     end
-    if realmatrix
-        TT = typeof(sqrt(zero(T)))
-    else
-        TT = typeof(sqrt(complex(-one(T))))
-    end
-    R = zeros(TT, n, n)
-    for j = 1:n
-        R[j,j] = realmatrix?sqrt(A[j,j]):sqrt(complex(A[j,j]))
+    sqrtm(A,Val{realmatrix})
+end
+# solve the sylvester equation a*x + x*b + c for x when a,b,x are commutative numbers. PR#20214
+sylvester(a::Union{Real,Complex},b::Union{Real,Complex},c::Union{Real,Complex}) = -c / (a + b)
```

Comment:
Hmm, it looks like LLVM is smart enough to eliminate the double negation either way:
```jl
julia> syl(a,b,c) = -c / (a + b)
syl (generic function with 1 method)

julia> foo(a,b,c) = syl(a,b,-c)
foo (generic function with 1 method)

julia> @code_llvm foo(1.0,2.0,3.0)

define double @julia_foo_65538(double, double, double) #0 !dbg !5 {
top:
  %3 = fadd double %0, %1
  %4 = fdiv double %2, %3
  ret double %4
}
```